sources/pmwebapi.go

```diff
@@ -110,8 +110,29 @@ func typeLabel(units string, pcpType string, name string) string {
 	} else {
 		name += unit
 	}
+	return name
+}
+
+func fixNaming(name string) string {
 	name = strings.Replace(name, "count / sec", "count_per_second", -1)
+	name = strings.Replace(name, "Kbyte", "kilobyte", -1)
+	name = strings.Replace(name, "nanosec", "nanoseconds", -1)
+
+	name = strings.Replace(name, "millisec", "milliseconds", -1)
+	name = strings.Replace(name, "failcnt", "failcount", -1)
+	name = strings.Replace(name, "Mbyte", "megabyte", -1)
+	name = strings.Replace(name, "byte", "bytes", -1)
+	name = strings.Replace(name, "sec", "seconds", -1)
+	name = strings.Replace(name, "bytes_byte", "bytes", -1)
+	name = strings.Replace(name, "bytess_bytes", "bytes", -1)
+	name = strings.Replace(name, "megabytesss", "megabytes", -1)
+	name = strings.Replace(name, "kilobytesss", "kilobytes", -1)
+	name = strings.Replace(name, "millisecondsonds", "milliseconds", -1)
+	name = strings.Replace(name, "nanosecondsonds", "nanoseconds", -1)
```

Comment:
Aren't these supposed to be "nanosecondonds" instead of "nanosecondsonds" (and so on)? I still feel like there's a smarter way to do this than making multiple fixes on the same metric to fix what the first fix breaks for some metrics. Any good ideas here, @roclark ?

Reply:
There's a bigger issue here. Look at the following example:
Say we have a metric name which is simple: `milliseconds` - there is nothing wrong with this name. Going through your chain, it will hit line 121 and replace `millisec` with `milliseconds`, giving you `millisecondsonds`. Then, line 125 will match again because it would replace `sec` with `seconds`, technically giving you `millisecondsondsonds`. Finally, line 130 updates this to `millisecondsonds` for your final answer. So... this actually still doesn't work properly. You can verify by adding another unit test case where you test `milliseconds` and it will throw an error saying that it retrieved `millisecondsonds`.

Long story short, if you end up using this method, you will need to think critically about the order of operations. Since `sec` is in the middle, and `sec` is in multiple names down below, it will keep updating itself. One way is to work on the lowest common denominators first (such as `sec` and `byte`) and then update the other ones later.

Also, remember to look at the worst case scenario for your unit tests. Yes, testing the specific names seen here should always work, but don't forget to test things that _aren't_ listed here that might not work. As I mentioned, even `milliseconds` will give issues.

Reply:
I will keep thinking on this and see if there is a more efficient way to catch these, but at the very least, what I mentioned so far is a good start.